src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpSamDAO.scala

```diff
@@ -214,8 +219,16 @@ class HttpSamDAO(implicit
     callback.future.map(_ => ())
   }
 
+  override def getUserStatus(user: WithAccessToken): Future[UserStatusInfo] = {
+    val apiClient = newApiClient(user)
+    val sam = new UsersApi(apiClient)
+    val callback = new SamApiCallback[UserStatusInfo]("getUserStatusInfo")
+    sam.getUserStatusInfoAsync(callback)
+    callback.future
+  }
+
   private def newApiClient(user: WithAccessToken) = {
-    val apiClient = new ApiClient()
+    val apiClient = new ApiClient(httpClient)
```

Comment:
This is the important change. `httpClient` is reused across all sam calls using the client library